torch/onnx/_internal/exporter/_compat.py

```diff
@@ -98,25 +101,18 @@ def _from_dynamic_axes_to_dynamic_shapes(
 
     dynamic_shapes: dict[str, Any | None] = {}
     for input_name, axes in dynamic_axes.items():
-        # NOTE: torch.export.Dim requires strict min and max constraints, and it
-        # dpends on the traced model to provide the correct min and max values.
-        # We set max to 99999 to avoid the constraints violation error with the default int64 max.
-        # https://github.com/pytorch/pytorch/blob/32f585d9346e316e554c8d9bf7548af9f62141fc/test/export/test_export.py#L687
+        # TODO(titaiwang): Add ONNX IR pass to post-process the dynamic axes
+        # torch.export.Dim.AUTO does its best to infer the min and max values
+        # from the model, but it's not guaranteed to be dynamic.
         if input_name in output_names:
             # User specified an output name as a dynamic axis, so we skip it
             continue
         if isinstance(axes, dict):
-            # Dim needs to pass str.isidentifier()
-            # If the max is not set, llm is going to fail, as sequence length is usually bounded within config.
-            # But we also don't want to only support llm. This kind of leaves us with this awkward position.
             dynamic_shapes[input_name] = {
-                k: torch.export.Dim(re.sub(r"[^A-Za-z_]", "", v), max=99999)
-                for k, v in axes.items()
+                k: torch.export.Dim.AUTO for k, _ in axes.items()
             }
         elif isinstance(axes, list):
-            dynamic_shapes[input_name] = {
-                k: torch.export.Dim(f"{input_name}_dim_{k}", max=99999) for k in axes
-            }
+            dynamic_shapes[input_name] = {k: torch.export.Dim.AUTO for k in axes}
```

Comment:
May be worth checking that `k` is a valid value (int in the expected range)?

Reply:
Done